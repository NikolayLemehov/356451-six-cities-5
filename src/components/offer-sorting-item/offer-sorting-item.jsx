import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {setSortedType, setSortedCityOffers} from "../../store/action";

const OfferSortingItem = (props) => {
  const {type, onCloseMenu, currentSortType, onChangeSortedType, onChangeSortedCityOffers} = props;

  return (
    <li className={`places__option ${currentSortType === type && `places__option--active`}`} tabIndex="0"
      onClick={() => {
        onChangeSortedType(type);
        onChangeSortedCityOffers();
        onCloseMenu();
      }}
    >{type}</li>
  );
};

OfferSortingItem.propTypes = {
  type: PropTypes.string.isRequired,
  currentSortType: PropTypes.string.isRequired,
  onChangeSortedType: PropTypes.func.isRequired,
  onChangeSortedCityOffers: PropTypes.func.isRequired,
  onCloseMenu: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  currentSortType: state.currentSortType,
});

const mapDispatchToProps = (dispatch) => ({
  onChangeSortedType(sortedType) {
    dispatch(setSortedType(sortedType));
  },
  onChangeSortedCityOffers() {
    dispatch(setSortedCityOffers());
  },
});

export {OfferSortingItem};
export default connect(mapStateToProps, mapDispatchToProps)(OfferSortingItem);
