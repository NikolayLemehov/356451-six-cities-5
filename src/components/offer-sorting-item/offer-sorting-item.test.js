import React from "react";
import renderer from "react-test-renderer";
import {OfferSortingItem} from "./offer-sorting-item";

const noop = () => {};
describe(`OfferSortingItem should render correctly`, () => {
  it(`OfferSortingItem should render correctly when the type is current`, () => {
    const tree = renderer
      .create(
          <OfferSortingItem
            type={`type1`}
            onCloseMenu={noop}
            currentSortType={`type1`}
            onChangeSortedType={noop}
          />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`OfferSortingItem should render correctly when the type is not current`, () => {
    const tree = renderer
      .create(
          <OfferSortingItem
            type={`type2`}
            onCloseMenu={noop}
            currentSortType={`type1`}
            onChangeSortedType={noop}
          />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
