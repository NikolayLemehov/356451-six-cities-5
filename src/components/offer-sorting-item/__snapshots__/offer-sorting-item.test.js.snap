// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferSortingItem should render correctly OfferSortingItem should render correctly when the type is current 1`] = `
<li
  className="places__option places__option--active"
  onClick={[Function]}
  tabIndex="0"
>
  type1
</li>
`;

exports[`OfferSortingItem should render correctly OfferSortingItem should render correctly when the type is not current 1`] = `
<li
  className="places__option"
  onClick={[Function]}
  tabIndex="0"
>
  type2
</li>
`;
